dfc/keepalivetracker.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+ */
+
+package dfc
+
+// This file has different implementations of KeepaliveTracker.
+// ideally this file stays under dfcpub, but currently dfc has a flat structure, following that
+// for now.
+import (
+	"time"
+)
+
+// HeartBeatTracker tracks timestamp of the last a message is received from a server.
+// Timeout: a message is not received within the interval.
+type HeartBeatTracker struct {
+	ch       chan struct{}
```

Comment:
Any reason you prefer this over `sync.Mutex`?

Reply:
avoid lock as much as possible, that's what i believe